onboarding/source/amb_light_service.c

```diff
@@ -18,22 +25,57 @@ static void lightServiceTask(void * pvParameters);
 uint8_t initLightService(void) {
     /* USER CODE BEGIN */
     // Create the task and queue here.
+    BaseType_t xReturned = pdFAIL;
+
+    // Create light task
+    if (lightServiceHandle == NULL) {
+        xReturned = xTaskCreate(lightServiceTask,
+                                LIGHT_SERVICE_NAME,
+                                LIGHT_SERVICE_STACK_SIZE,
+                                NULL,
+                                LIGHT_SERVICE_PRIORITY,
+                                &lightServiceHandle);
+    }
+    
+    if (xReturned == pdFAIL) {
+            printf("\nlightServiceTask not created...\n");
+    }
+
+    // Create queue
+    if (lightServiceQueue == NULL) {
+        lightServiceQueue = xQueueCreate(LIGHT_QUEUE_LENGTH, LIGHT_QUEUE_LENGTH);
+    }
 
     /* USER CODE END */
     return 1;
 }
 
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
+    adcData_t adc_data;
+    adcData_t *adc_data_ptr = &adc_data;
+
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
+    while (1) {
+        if (xQueueReceive(lightServiceQueue, NULL, (TickType_t) 11) == pdPASS) {
```

Comment:
Try not to use magic numbers. Instead, define a macro for this block time or use `portMAX_DELAY`